pkg/operation/seed/seed.go

```diff
@@ -563,7 +563,7 @@ func RunReconcileSeedFlow(
 			if err != nil {
 				return err
 			}
-			if len(currentResources) != 0 && currentResources[resource] != nil {
+			if len(currentResources) != 0 && currentResources["prometheus"] != nil {
 				monitoringResources[resource] = map[string]interface{}{
 					resource: currentResources[resource],
```

Comment:
```suggestion
					"prometheus": currentResources[resource],
```
Right?

Reply:
I fixed it now. `resource` should not be changed because it is the key for the values https://github.com/gardener/gardener/blob/master/pkg/operation/seed/seed.go#L781
`currentResources[resource]` should always be `currentResources["prometheus"]` because the container name is always `prometheus`.